backend/src/routes/auth.ts

```diff
@@ -160,4 +165,121 @@ export const authRoute = new Elysia({ prefix: '/auth' })
             status: 'success',
             message: `${user.email} has been verified`,
         };
+    })
+    .post('/sign-in', async ({ request, cookie }) => {
+        const formData = await request.formData();
+        const validEmaillPass = signInFormSchema.safeParse({
+            email: formData.get('email'),
+            password: formData.get('password'),
+        });
+
+        if (!validEmaillPass.success) {
+            return {
+                status: 'error',
+                errors: validEmaillPass.error.flatten().fieldErrors,
+            };
+        }
+
+        const { email, password } = validEmaillPass.data;
+
+        let user_id = '';
+
+        try {
+            const queryAuthUserData = await sql`
+            SELECT id, hashed_password
+            FROM auth_user
+            WHERE email = ${email}
+            `;
+
+            const queriedHashedPassword = queryAuthUserData[0].hashed_password;
+            user_id = queryAuthUserData[0].id;
+
+            const isPasswordMatch = await new Argon2id().verify(
+                queriedHashedPassword,
+                password,
+            );
+
+            if (!isPasswordMatch) {
+                return {
+                    status: 'error',
+                    response: {
+                        message: 'email or password is incorrect',
+                    },
+                };
+            }
+        } catch (error: any) {
+            return {
+                status: 'error',
+                response: {
+                    message: error,
```

Comment:
RETURN SERVER ERROR TO CLIENT SIDE IS EXTREMLY DANGEROUS!
instead check for type of error and handle it with easy to understand message